src/app/components/website-status/website-status.component.ts

```diff
@@ -0,0 +1,112 @@
+  template: `
+    <h2>Website Status</h2>
+    <baw-items [items]="statusItems()"></baw-items>
+
+    <p>
+      If you are experiencing issues with the website, please
+      <a [strongRoute]="reportProblemRoute">Report a Problem</a> so we can
+      investigate.
+    </p>
+  `,
+})
+class WebsiteStatusComponent extends PageComponent implements OnInit {
+  public constructor(private api: WebsiteStatusService) {
+    super();
+  }
+
+  protected reportProblemRoute = reportProblemMenuItem.route;
+
+  public ngOnInit(): void {
+    this.api
+      .show()
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe((status) => {
+        this.model = status;
+      });
+  }
+
+  protected model: WebsiteStatus;
+
+  protected statusItems(): List<IItem> {
+    // if we get no response from the server, the model will be "null"
+    // if we want to see if the server connection is healthy, we want to negate this.model?.timedOut
+    // (so that true) means the server did not time out
+    // however, because "null" is a falsey value, negating "null" will return true (representing that the server connection is healthy)
+    // if we didn't receive a response from the server, we want to keep "null" so that we can display "Unknown" in the template
+    // we therefore need to check if the model is instantiated before we negate it
+    const serverTimeout: boolean = isInstantiated(this.model?.timedOut)
+      ? !this.model.timedOut
+      : null;
+
+    return List<IItem>([
+      {
+        name: "Overall Server Health",
+        value: this.healthyUnhealthyText(this.model?.isStatusHealthy),
+        icon: ["fas", "heartbeat"],
+      },
+      {
+        name: "Server Connection",
+        value: this.healthyUnhealthyText(serverTimeout),
+        icon: ["fas", "network-wired"],
+      },
+      {
+        name: "Database",
+        value: this.healthyUnhealthyText(this.model?.database),
+        icon: ["fas", "database"],
+      },
+      {
+        name: "Cache",
+        value: this.healthyUnhealthyText(this.model?.isRedisHealthy),
+        icon: ["fas", "memory"],
+      },
+      {
+        name: "Storage",
+        value: this.healthyUnhealthyText(this.model?.isStorageHealthy),
+        icon: ["fas", "hdd"],
+      },
+      {
+        name: "User Uploads",
+        value: this.healthyUnhealthyText(this.model?.isUploadingHealthy),
+        icon: ["fas", "cloud"],
+      },
+      {
+        name: "User Internet Connection",
+        value: this.healthyUnhealthyText(navigator.onLine),
+        icon: ["fas", "wifi"],
+      },
+    ]);
+  }
+
+  private healthyUnhealthyText(healthy?: boolean): string {
+    // if we don't receive a response from the sever (the sever is down) or the user doesn't have internet
+    // the "healthy" field will be "null". In this case, we want to display "Unknown" in the template
+    if (!isInstantiated<boolean>(healthy)) {
+      return "Unknown";
+    }
+
+    return healthy ? "Healthy" : "Unhealthy";
```

Comment:
semantic colors of the badges might look ok?

Reply:
Updated screenshot with semantic colours